app/src/main/java/org/mozilla/focus/utils/UrlUtils.java

```diff
@@ -62,6 +64,74 @@ public static String createSearchUrl(Context context, String searchTerm) {
         return searchEngine.buildSearchUrl(searchTerm);
     }
 
+    /**
+     * @return The search terms for the first search performed with the default search engine or the current URL.
+     */
+    public static String getSearchTermsOrUrl(Context context, String url) {
+        String searchTermsOrUrl = null;
+
+        if (UrlUtils.isDefaultSearchUrl(context, url)) {
+            searchTermsOrUrl = UrlUtils.getSearchTermsFromUrl(context, url);
+        }
+
+        if (searchTermsOrUrl == null || searchTermsOrUrl.isEmpty()) {
+            searchTermsOrUrl = url;
+        }
+
+        return searchTermsOrUrl;
+    }
+
+    /**
+     * @return If a url matches the form of a first search URL for the current default search engine.
+     */
+    private static boolean isDefaultSearchUrl(Context context, String urlSearch) {
```

Comment:
\[PMD\] INFO: The method 'isDefaultSearchUrl' has a Cyclomatic Complexity of 6\.